activation/activation.go

```diff
@@ -38,26 +38,26 @@ func DefaultPoetConfig() PoetConfig {
 
 const defaultPoetRetryInterval = 5 * time.Second
 
-type nipostBuilder interface {
+type NipostBuilder interface {
```

Comment:
is it possible to move these interface inside interface.go?

btw, why do they need to be public interface?

Reply:
Good point. These interface should be private.

Regarding moving them to `interface.go`: I'd prefer getting rid of `interface.go` and move the interfaces to where they are used. The only code that depends on that interface is inside this file. Yes, the actual implementation of this interface is in `nipost.go` (or for the mock in `activation_mocks.go`), but these don't depend on the interface, they just implement it.

Reply:
for me, within each package, the interface is already residing in the place they are used, either in interface.go or in each xxxxx.go file. the two advantage i see:
- one go:generate directive to generate the mock
- sometimes multiple go files use the same interface

Reply:
I moved (most) interfaces to `interface.go` and made them private.